src/core/lib/security/credentials/tls/grpc_tls_certificate_verifier.h

```diff
@@ -57,19 +61,20 @@ struct grpc_tls_certificate_verifier
   // Compares this grpc_tls_certificate_verifier object with \a other.
   // If this method returns 0, it means that gRPC can treat the two channel
   // credentials as effectively the same.
-  int cmp(const grpc_tls_certificate_verifier* other) const {
+  int Compare(const grpc_tls_certificate_verifier* other) const {
     GPR_ASSERT(other != nullptr);
     int r = strcmp(type(), other->type());
     if (r != 0) return r;
-    return cmp_impl(other);
+    return CompareImpl(other);
   }
 
   const char* type() const { return type_; }
```

Comment:
As we've discussed offline, let's change the API for all of the classes involved here (credentials, verifier, provider, and anything else) such that instead of passing the type name pointer to the base class and requiring the base class to store it, we instead just make the `type()` method pure virtual, so that we can directly inline the string in that method in the subclasses.

Reply:
Done